packages/amplify_core/lib/src/util/get_base_element_href_from_dom.web.dart

```diff
@@ -0,0 +1,9 @@
+import 'dart:html';
+
+import 'package:amplify_core/amplify_core.dart';
+
+/// {@macro amplify_core.get_base_element_href_from_dom}
+String? getBaseElementHrefFromDom() {
+  if (!zIsFlutter) return null;
+  return document.querySelector('base')?.getAttribute('href');
```

Comment:
Is this necessary to split out?  Seems like a lot of mental overhead to have the import from core, 3 files to 'support' all platforms, and then an import into the db package for what could be 1-2 lines in the db package itself?

Reply:
We probably don't need three files since this would never be imported in a shared library but only from an already web-only library.

Reply:
I did 3 so that it could be imported from `package:amplify_core/amplify_core.dart`, but yeah, we can drop it to one if we want to have the import from `package:amplify_core/src/util/get_base_element_href_from_dom.dart`

Or we create a `package:amplify_core/amplify_core_web.dart` that exports anything web specific.

We just can't put anything web specific in amplify_core.dart.

Reply:
I went the route of creating `package:amplify_core/web.dart`. I think that is the typical way this would be handled.

Reply:
Sounds good, although if it can be used in a Dart-only context, I would rather add it to `package:aws_common/web.dart`